sentry-apache-http-client-5/src/test/kotlin/io/sentry/transport/apache/ApacheHttpClientTransportTest.kt

```diff
@@ -100,4 +108,45 @@ class ApacheHttpClientTransportTest {
         sut.close()
         verify(fixture.client).close(CloseMode.GRACEFUL)
     }
+
+    @Test
+    fun `flush waits till all requests are finished`() {
```

Comment:
can we add a test that we flush X items and if we enqueue 1 more later, its also sent? so we guarantee that the `ReusableCountLatch` is actually, reusable.

Reply:
Added